ChainSafe/gossamer
# feat(telemetry): send txpool.import telemetry msg

## Changes



- From this commit, a telemetry message of type `txpool.import` will be
sent when a new transaction is imported in the transaction pool.

- Added tests for `txpool.import` and improved TestHandler_SendMulti test.

## Tests



- Setup a local telemetry server using substrate-telemetry.

Use the branch `print-payload-type` from my fork `https://github.com/kishansagathiya/substrate-telemetry` (It print the message received, type of message and error. From what I have played around upstream doesn't do that.)

Run telementry backend (`telemetry_core` and `telemetry_shard`).

-  Run gossamer by setting telemetry  url to the local server. 

Run `./bin/gossamer --chain polkadot --name 'Kishans Test Node' --telemetry-url 'ws://localhost:8001/submit 0'`

Follow the logs of telemetry shard. Wait for few minutes. Eventually, you should see logs like below
```
telemetry message received: {"best":"0x8f9e20e28e1a5b5fb1b9011e1b60a7fdf42fb610c0f4ebf6e5d9739ced7648ee","height":1645,"msg":"block.import","origin":"NetworkInitialSync","ts":"2021-11-09T18:34:05.96264119+05:30"}
message type Payload::BlockImport

```

## Issues



- Closes #1836

## Primary Reviewer



- @noot
